### ISSUE TITLE
Implementic automatic releases (?)

### ISSUE THREAD

**Issue by smoia** [ME-ICA/tedana#571]
### Summary
Following today's call (#569) and a discussion started in PR #568, there are tools to create automatic releases when an event take place (for instance, when a PR is merged into `main`).

One of such tools is [auto](https://github.com/intuit/auto).
Auto creates automatically a release at every PR, preparing the release file and the log using the PR title (just like release-drafter). It also automatically update the version based on pre-setted information. It's originally based on labels, but it also supports conventional commits (that follow better the current guidelines for PR). There are also other options out there, some specifically built for python!

In general, a solution like this have pros and cons.
Pros:
  + It automatises some steps, reducing management work and _babysitting_.
  + Users could have access to the latest functionality from pypi, and in any case from a semantic versioned release.
  + It automatises semantic versioning taking away feelings from it.
  + Releases result "made" by a user rather than by a bot (like release-drafter).
  + In one step and one tool, it does what `release-drafter`, `all-contributors`, and a developer would do.
  + It can be set so that there is a `stable` branch, a `latest` branch and eventually `alpha/beta/rc/...` branches. This means that a `latest` branch could have a release (that can be tagged as pre-release) at each PR merge, while a `stable` branch would have a release every time a maintainer merges in the `latest` branch (e.g. creating quarterly releases).

Cons:
  - Sometimes _babysitting_ is necessary, or just better than automation.
  - Like `release-drafter` (and not unlikely what is the practice now), it requires the PR initiator (and/or the PR reviewer) to adopt certain conventions, such as naming the PR in the right way or improving the explanation of the PR to make it user-friendly.
  - If it doesn't save much time, then it's not that helpful.

**Comment by jbteves** [comment]
I would like to expand on my objections to auto-drafting releases. There are a few things that I would like to raise concerns about:
- The "babysitting" is extremely helpful for us to understand which changes are more impactful to users or not.
- Automatic releasing means that we're just babysitting a program instead of the text for the release and the PyPi deploy, and could easily cost a significant amount of time to set up and maintain.
- The program cannot distinguish things into categories to help our users understand what types of changes belong together (e.g., bug fixes vs. enhancements) unless the PR correctly identifies it.
- I am reluctant to place more burdens on reviewers and contributors, rather than burdening one person or two for some time before a release. The cost in person-hours of checking the PRs more heavily is much higher.

**Comment by tsalo** [comment]
I agree with @jbteves' points for the most part. I think that the burden of reorganizing and summarizing the changes at the time of a given release are lower than the burden of ensuring that all PRs match the new requirements to be informative. That said, I think `auto` sounds like a very cool tool and I wonder what others think about it.

**Comment by eurunuela** [comment]
Given that we're planning on merging `auto` into phy2bids soon, I'd say we wait and see how it goes. Some of us will have had some experience with the tool and we'll know if it's worth it.

**Comment by emdupre** [comment]
I'd agree that it's good to get more information on its usage in another project ! My gut is against using this for the reasons @jbteves mentions. I also think we don't maintain a `stable` branch, which this seems particularly well-suited for, so we'd need to significantly update our development workflow.

Maybe we can ask for an update on `auto` in phys2bids in the next month or two ?

**Comment by jbteves** [comment]
I guess I'll add in that while I don't agree with its use in `tedana` for the reasons outlined, there is clear community interest and I can see the utility of such a tool. For that reason, it might be prudent to facilitate a community discussion on how this tool might benefit neuroimagers, maybe with some notes from `phys2bids` contributors on how well it worked for that project. We definitely have access to a variety of developers and communities here, all of whom might benefit. So I don't want to accidentally squash discussion on `auto`'s broader use, because I sense a lot of excitement about its possibilities. I hope that clarifies my take!

**Comment by eurunuela** [comment]
Cool. @smoia, @rmarkello and I will get back to you then with our experience with `auto` once we merge it in `phys2bids`.

**Comment by eurunuela** [comment]
@smoia finally made it work on phys2bids! 🎉 🍾

We can continue with this discussion (if necessary) once we see it work in phys2bids for a few weeks.

**Comment by tsalo** [comment]
I think we're all pretty happy with the current approach, so I'm going to close this and we can revisit if automated releases become a priority.
